LocalCameraController rotate camera around character when not visible

SAMPROO opened the pull request:
- If camera ray hits a wall, rotate camera around the character until the character is visible
- Ignore animator controller (not sure how it ended up in here. It has its own repo)

SAMPROO committed:
Animator Controller

- Simple version of animation controller for future animations
- Animator to character prefabs
- LocalController uses animator controller to let controller know which animation
- Animation speed multiplier

SAMPROO committed:
Merge branch 'master' into AnimatorController

SAMPROO committed:
Fixed new LocalCharcterController script compatibility with AnimatorController

- Removed old enable movement script (this will be instead done with animation)
- Added movement state to LocalCharacterController

SAMPROO committed:
Deleted comment

SAMPROO committed:
Current movement state interpolate

- added sidewaysRun and backwardsRun for LocalCameraController

SAMPROO committed:
When running camera goes forward acording to the speed of run

SAMPROO committed:
Camera look ahead without LookAt() (unfinished)

SAMPROO committed:
Camera moves upwards and forwards when character is behind "Wall"

SAMPROO committed:
Fixed quick bug

SAMPROO committed:
Created method for moving camera when character is behind a wall

SAMPROO committed:
Camera default position change

SAMPROO committed:
Removed camera from test prefab

SAMPROO committed:
Merge branch 'master' into LocalCameraController

SAMPROO committed:
Merge branch 'master' into LocalCameraController

SAMPROO committed:
Temporary fix for moving camera back to original position

- fixed bug where camera is always dragging behind character when moving fast by adding speed to moving the camera back

- removed some weird character from LocalPlayerController

SAMPROO committed:
New simpledfied and improved MoveCameraIfCharacterNotVisible

- Camera positioin calculated using offset from target, distance and angle
- Check if character visible using two rays for prevention of jittering camera movement
- Increment camera angle around character if character not visible from default position
- Decrement camera angle around character if character visible from checkup ray, which is under default position ray (clamp angle between min and max)
- Created method for checking if character is visible from from position
- Created method for getting camera postion using target, distance and angle

SAMPROO committed:
Merge branch 'master' into LocalCameraController

Iiropel commented on MorkoUnityProject/Assets/Code/LocalCameraController.cs:
```diff
@@ -3,18 +3,77 @@
+		return position;
+	}
+
+	private Vector3 MoveCameraIfCharacterNotVisible(Vector3 currentPosition, float minAngle, float maxAngle)
```
Funktion nimi ei vastaa tapahtumia, liikuttaminen tapahtuu näkyi tai ei. Se ei myöskään liikuta vaan antaa pelkästään position. Muuttujien nimet myös hieman vaikeita ymmärtää.

Iiropel commented on MorkoUnityProject/Assets/Code/LocalCameraController.cs:
```diff
@@ -3,18 +3,77 @@
+
+	private bool IsTargetVisible(Vector3 from, Vector3 target, string hitTag = "Wall")
+	{
+		var forward = target - from;
```
Muuta tää funktio ottamaan arrayn kaikista raycastin varrella olevista collidereista. Sitten voidaan itse määrittää mitkä asiat triggeröivät kameran kulman muutoksen.

Forward mielummin -> dir tai direction.
From on myös vähän outo.

Iiropel commented on MorkoUnityProject/Assets/Code/LocalPlayerController.cs:
```diff
@@ -136,34 +211,54 @@ private void Move(Vector3 moveDirection, bool accelerateRun, bool hasMoved)
 		else if (maxRunSpeed)
 		{
+			currentMovementState = MovementState.Run;
 			ran = true;
```
ran?

Iiropel commented on MorkoUnityProject/Assets/Code/LocalPlayerController.cs:
```diff
@@ -180,26 +275,39 @@ private void Move(Vector3 moveDirection, bool accelerateRun, bool hasMoved)
 		
-		bool movingSideways = moveLookDotProduct >= 0 && moveLookDotProduct < 1f;
-		bool movingBackwards = moveLookDotProduct < 0 && moveLookDotProduct >= -1f;
+		bool movingSideways = moveLookDotProduct >= -0.32 && moveLookDotProduct <= 0.66f;
```
Onko nää tarkotus jättää kovakoodatuiksi arbitrary numeroiksi?

Iiropel reviewed (commented):
Pari juttua

SAMPROO committed:
Updated LocalCameraController

- Fixed MoveCameraIfCharacterNotVisible naming to GetNewCameraPositionIfCharacterNotVisible

- IsTargetVisible now takes in an array of tags

- varibale naming changes
